scalameta/tokenizers/shared/src/main/scala/scala/meta/internal/tokenizers/LegacyScanner.scala

```diff
@@ -332,11 +332,17 @@ class LegacyScanner(input: Input, dialect: Dialect) {
         if (ch == '$' && !getDollar()) {
           getUnquote()
         } else {
+          val prevChar = ch
           putChar(ch)
           nextChar()
-          getIdentRest()
-          if (ch == '"' && token == IDENTIFIER)
-            token = INTERPOLATIONID
+          if (prevChar == '$' && ch == '{' && dialect.allowWhiteboxMacro) {
```

Comment:
How does it deal with things like `" This is the ${value}"` ?

Reply:
a little above you have:
`    if (inStringInterpolation) return fetchStringPart()`
basically this cose is not called when in string interpolation